### PR TITLE
Jhe java8mig178

### PR DESCRIPTION
**Pull request by apgsga-jhe**
Check module names: updated as discussed, should be better now.

### CODE REVIEW

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 1]
```diff
@@ -215,5 +222,60 @@ protected void setJenkinsClient(JenkinsPatchClient jenkinsClient) {
...
-
+	
+	private List<MavenArtifact> getArtifactsWithNameFromBom(String version) {
+		List<MavenArtifact> artifactsWithNameFromBom = null;
+		try {
+			artifactsWithNameFromBom = am.getArtifactsWithNameFromBom(version);
+		}
+		catch(Exception ex) {
+			//TODO JHE: do we want to do anything special here?
+			System.out.println(ex.getMessage());
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Yep, throw a Runtime Exception

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 2]
```diff
@@ -215,5 +222,60 @@ protected void setJenkinsClient(JenkinsPatchClient jenkinsClient) {
...
+		}
+		
+		return artifactsWithNameFromBom;
+	}
+	
+	private List<MavenArtifact> getArtifactNameError(List<MavenArtifact> mavenArtifacts, String cvsBranch) {
+		
+		VcsCommandRunner cmdRunner = initAndGetVcsCommandRunner();
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Just a comment, no follow - up needed: why here an own method initAndGetVcsCommandRunner for create and preprocess?

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 3]
```diff
@@ -215,5 +222,60 @@ protected void setJenkinsClient(JenkinsPatchClient jenkinsClient) {
...
+		List<MavenArtifact> artifactWihInvalidNames = Lists.newArrayList();
+		
+		for(MavenArtifact ma : mavenArtifacts) {
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: As a thought , also no immediate follow-up needed. Maybe using Java8 Stream with maps and filters would make the code shorter here?

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java [hunk 4]
```diff
@@ -215,5 +222,60 @@ protected void setJenkinsClient(JenkinsPatchClient jenkinsClient) {
...
+					List<String> cvsResults = cmdRunner.run(silentCoCmd);
+					// JHE: SilentCOCvsModuleCommand returns 0 when all OK, 1 instead...
+					if(cvsResults.size() != 1 || cvsResults.get(0).equals("1")) {
+						artifactWihInvalidNames.add(ma);
+					}
+				}
+			}
+			catch(Exception ex) {
+				//TODO JHE: do we want to do anything special here?
+				System.out.println(ex.getMessage());
```
Comment (chhex) [inline apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Again, here we need proper Logging and the rethrowing of the Exception as a Runtime Exception. 
Logging: Ideally Log also the Stacktrace withh Commons Exception Utils.

**Comment by chhex** [comment]
I ll let you do the merge?  
Can you give a short mail, when you merge?

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java (hunk 2 above)
Comment (apgsga-jhe) [reply apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Good question :/ ... I don't remember, probably thought at the beginning that one needs to do more to init it ... anyway, I removed this method now and do everything within getArtifactNameError method.

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java (hunk 1 above)
Comment (apgsga-jhe) [reply apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Done.

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java (hunk 3 above)
Comment (apgsga-jhe) [reply apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Good point. As you said I won't change this now.

File: apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java (hunk 4 above)
Comment (apgsga-jhe) [reply apg-patch-service-server/src/main/java/com/apgsga/microservice/patch/server/impl/SimplePatchContainerBean.java]: Done, yep with commons exception utils.

**Comment by apgsga-jhe** [comment]
Yep sure, Ill do the minor changes based on your review, commit, merge, and let you know.

**Pull request merged**